test/util/xcm.ts

```diff
@@ -919,3 +919,43 @@ export const expectXcmEventMessage = async (context: DevModeContext, message: st
 
     return filteredEvents.length ? filteredEvents[0]!.data.error.toString() === message : false;
 };
+
+export const extractPaidDeliveryFees = async (context: DevModeContext) => {
+    const records = await context.polkadotJs().query.system.events();
+
+    const filteredEvents = records
+        .map(({ event }) => (context.polkadotJs().events.polkadotXcm.FeesPaid.is(event) ? event : undefined))
+        .filter((event) => event);
+
+    return filteredEvents[0]!.data[1][0].fun.asFungible.toBigInt();
+};
+
+export const getLastSentUmpMessageFee = async (context: DevModeContext, baseDelivery: bigint, txByteFee: bigint) => {
+    const sentXcm = (await context.polkadotJs().query.parachainSystem.upwardMessages())[0];
```

Comment:
Could this be a runtime error if the message list is empty?

Reply:
If the message list is empty it will simply throw a typescript error.

Reply:
the runtime will return an empty vec and you will try to get the item 0, which does not exist

Reply:
yes, I mean typescript error. Do we safeguard against that? Or it is fine?